src/plugins/data_explorer/public/utils/state_management/metadata_slice.ts

```diff
@@ -26,9 +26,18 @@ export const getPreloadedState = async ({
       .getStateTransfer(scopedHistory)
       .getIncomingEditorState({ keysToRemoveAfterFetch: ['id', 'input'] }) || {};
   const isQueryEnhancementEnabled = uiSettings.get(QUERY_ENHANCEMENT_ENABLED_SETTING);
-  const defaultIndexPattern = isQueryEnhancementEnabled
-    ? undefined
-    : await data.indexPatterns.getDefault();
+
+  let defaultIndexPattern;
```

Comment:
generally speaking https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#avoid-mutability-and-state

is this the right solution here? 

if query enhancements is disabled and there is no default index pattern. couldn't we just redirect here?